net/src/main/java/org/apollo/net/update/resource/WorldListResourceProvider.java

```diff
@@ -19,25 +17,33 @@ public boolean accept(String path) throws IOException {
 
 	@Override
 	public Optional<ByteBuffer> get(String path) throws IOException {
-		final var buf = Unpooled.buffer();
 		final var world = createWorld();
+		final var buf = ByteBuffer.allocate(Integer.BYTES + world.remaining());
 
-		buf.writeInt(world.readableBytes());
-		buf.writeBytes(world);
+		buf.putInt(world.remaining());
+		buf.put(world);
 
-		return Optional.of(buf.nioBuffer());
+		return Optional.of(buf.flip());
 	}
 
-	private ByteBuf createWorld() {
-		ByteBuf world = Unpooled.buffer();
-		world.writeShort(1); // number of worlds
-
-		world.writeShort(1); // world id
-		world.writeInt(0); // world settings
-		BufferUtil.writeString(world, "127.0.0.1"); // address
-		BufferUtil.writeString(world, "Converting Apollo"); // address
-		world.writeByte(0); // country
-		world.writeShort(2000);// number of players
-		return world;
+	private ByteBuffer createWorld() {
```

Comment:
Do we actually need this to get login working?

Reply:
Client freezes without it when clicked on by mistake.

Reply:
Hm, then I'd be inclined to make it return real data. cc @Major-